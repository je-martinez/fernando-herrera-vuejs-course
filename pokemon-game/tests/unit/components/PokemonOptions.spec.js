import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions";
import { pokemonsMock } from "../mocks/pokemons.mock";

describe("PokemonOptions Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons: pokemonsMock,
      },
    });
  });

  test("should match with snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should show the 4 options correctly", () => {
    const [p1, p2, p3, p4] = wrapper.findAll("li");

    expect(p1.exists()).toBeTruthy();
    expect(p1.text()).toBe("bulbasaur");
    expect(p2.exists()).toBeTruthy();
    expect(p2.text()).toBe("pikachu");
    expect(p3.exists()).toBeTruthy();
    expect(p3.text()).toBe("mew");
    expect(p4.exists()).toBeTruthy();
    expect(p4.text()).toBe("caterpie");
  });
});
