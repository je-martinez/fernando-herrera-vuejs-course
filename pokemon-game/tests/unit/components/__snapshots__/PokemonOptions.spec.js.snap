// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component should match with snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>pikachu</li>
    <li>mew</li>
    <li>caterpie</li>
  </ul>
</div>
`;
